src/math/ChebyshevBoundedField.cc

```diff
@@ -427,15 +427,12 @@ PTR(BoundedField) ChebyshevBoundedField::operator*(double const scale) const {
 }
 
 bool ChebyshevBoundedField::operator==(BoundedField const& rhs) const {
-    try {
-        auto& rhsCasted = dynamic_cast<ChebyshevBoundedField const&>(rhs);
-        bool coefficientsEqual = _coefficients.getShape() == rhsCasted._coefficients.getShape() &&
-                                  all(equal(_coefficients, rhsCasted._coefficients));
-        return (getBBox() == rhsCasted.getBBox()) && coefficientsEqual;
-    }
-    catch (std::bad_cast&) {
-        return false;
-    }
+    auto rhsCasted = dynamic_cast<ChebyshevBoundedField const *>(&rhs);
+    if (!rhsCasted) return false;
```

Comment:
Is the "efficiency" just sidestepping the exception handler, or is there something more subtle going on?